### PR TITLE
feat: support 0.73 by removing compileOptions, kotlinOptions

### PR DESCRIPTION
**Pull request by yungblud**
Support RN 0.73 Android Native Build

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -39,16 +39,6 @@ android {
...
     }
   }
-
-  compileOptions {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
-  }
-
-  kotlinOptions {
-    freeCompilerArgs = ['-Xjvm-default=all']
```
Comment (jerielng) [inline android/build.gradle:49]: Thanks @yungblud - is there any reason you also want to remove `kotlinOptions.freeCompilerArgs`, or would it be sufficient for your app to remove just the strict Java dependencies to allow for Java 17?

Comment (yungblud) [reply android/build.gradle:49]: Hmm, let me test it.
I think if just removing compileOptions is good, that is the best way.

Comment (yungblud) [reply android/build.gradle:49]: Yeah, it's okay to remain freeCompilerArgs.
That was my bad.

But weirdly, I pushed onto this branch but github didn't apply it to PR.

Comment (yungblud) [reply android/build.gradle:49]: I will close this PR, and reopen it.
I think this is github's bug.

Comment (yungblud) [reply android/build.gradle:49]: https://github.com/braze-inc/braze-react-native-sdk/pull/242

**Pull request closed without merging**